gendiff/core.py

```diff
@@ -11,36 +11,57 @@ def get_keys_groups(
 
     union = keys1_set | keys2_set
     common = keys1_set & keys2_set
-    missing = keys1_set - common
 
     return {
         "union": union,
         "common": common,
-        "missing": missing,
+        "missing": keys1_set - common,
+        "new": keys2_set - common,
     }
 
 
-def get_diff_str(dict1: Dict[str, Any], dict2: Dict[str, Any]) -> str:
+def get_diff_view(dict1: Dict[str, Any], dict2: Dict[str, Any]) -> Dict[str, Any]:
     keys_groups = get_keys_groups(dict1, dict2)
-    output_str = "{\n"
+    diff_view = {}
+
+    for key in sorted(list(keys_groups["union"])):
+        if key in keys_groups["common"] and dict2[key] == dict1[key]:
+            diff_view[key] = "equal"
+        elif key in keys_groups["common"] and dict2[key] != dict1[key]:
+            if type(dict1[key]) == dict and type(dict2[key]) == dict:
+                diff_view[key] = get_diff_view(dict1[key], dict2[key])
+            else:
+                diff_view[key] = "common_diff"
+
+        elif key in keys_groups["missing"]:
+            diff_view[key] = "missing"
+        elif key in keys_groups["new"]:
+            diff_view[key] = "new"
+
+    return diff_view
 
-    for b in sorted(list(keys_groups["union"])):
-        if b in keys_groups["common"] and dict2[b] == dict1[b]:
-            output_str += f"      {b}: {dict2[b]}\n"
-        elif b in keys_groups["common"] and dict2[b] != dict1[b]:
-            output_str += f"    + {b}: {dict2[b]}\n"
-            output_str += f"    - {b}: {dict1[b]}\n"
-        elif b in keys_groups["missing"]:
-            output_str += f"    - {b}: {dict1[b]}\n"
-        else:
-            output_str += f"    + {b}: {dict2[b]}\n"
 
+def get_stylish_diff_format(dict1: Dict[str, Any], dict2: Dict[str, Any]) -> str:
+    diff_view = get_diff_view(dict1, dict2)
+    output_str = "{\n"
+    for key in diff_view.keys():
+        if diff_view[key] == "equal":
+            output_str += f"      {key}: {dict2[key]}\n"
+        elif diff_view[key] == "common_diff":
+            output_str += f"    - {key}: {dict1[key]}\n"
+            output_str += f"    + {key}: {dict2[key]}\n"
+        elif diff_view[key] == "missing":
+            output_str += f"    - {key}: {dict1[key]}\n"
+        elif diff_view[key] == "new":
+            output_str += f"    + {key}: {dict2[key]}\n"
+        else:
+            output_str += f"      {key}: \n"
+            output_str += f"{get_stylish_diff_format(dict1[key], dict2[key])}\n"
```

Comment:
your function doesn't know how deep it should add nested stuff, consider adding special parameter for padding/depth level that you'll increase with each recursive call